samples/WebsocketConnect/README.md

```diff
@@ -48,3 +48,139 @@ If you wish to use the latest SDK release to run the sample rather than using th
 ```sh
 mvn -P latest-release compile exec:java -pl samples/WebsocketConnect -Dexec.mainClass=websocketconnect.WebsocketConnect -Dexec.args="--endpoint <endpoint> --signing_region <signing region>"
 ```
+
+## Alternate connection configuration methods supported by AWS IoT Core
+
+### MQTT over WebSockets with static AWS credentials
+
+With a help of a static credentials provider your application can use a fixed set of AWS credentials. For that, you need
+to instantiate the `StaticCredentialsProviderBuilder` class and provide it with the AWS credentials. The following code
+snippet demonstrates how to set up an MQTT3 connection using static AWS credentials for SigV4-based authentication.
+
+```java
+static MqttClientConnection createMqttClientConnection(CommandLineUtils.SampleCommandLineData cmdData) {
+    try (AwsIotMqttConnectionBuilder builder = AwsIotMqttConnectionBuilder.newMtlsBuilderFromPath(null, null)) {
+        if (cmdData.input_ca != "") {
+            builder.withCertificateAuthorityFromPath(null, cmdData.input_ca);
+        }
+        builder.withConnectionEventCallbacks(callbacks)
+            .withClientId(cmdData.input_clientId)
+            .withEndpoint(cmdData.input_endpoint)
+            .withCleanSession(true)
+            .withProtocolOperationTimeoutMs(60000);
+
+        builder.withWebsockets(true);
+        builder.withWebsocketSigningRegion(cmdData.input_signingRegion);
+
+        StaticCredentialsProviderBuilder providerBuilder = new StaticCredentialsProviderBuilder();
+        providerBuilder.withAccessKeyId("<access key id>");
+        providerBuilder.withSecretAccessKey("<secret access key>");
+        providerBuilder.withSessionToken("<session>");
+
+        CredentialsProvider credentialsProvider = providerBuilder.build();
+        builder.withWebsocketCredentialsProvider(credentialsProvider);
+
+        MqttClientConnection connection = builder.build();
+        return connection;
+    }
+}
+```
+
+### MQTT over WebSockets with Custom Authorizer
+
+An MQTT3 direct connection can be made using a [Custom Authorizer](https://docs.aws.amazon.com/iot/latest/developerguide/custom-authentication.html).
+When making a connection to a Custom Authorizer, the MQTT3 client can optionally passing username, password, and/or token
+signature arguments based on the configuration of the Custom Authorizer on AWS IoT Core.
+
+You will need to setup your Custom Authorizer so that the lambda function returns a policy document to properly connect.
+See [this page](https://docs.aws.amazon.com/iot/latest/developerguide/config-custom-auth.html) on the documentation for
+more details and example return results.
+
+If your Custom Authorizer does not use signing, you don't specify anything related to the token signature and can use
+the following code:
+
+```java
```

Comment:
Same input as above snippet. Simplify by removing anything that's optional/unnecessary to establish this connection. Remove all references to cmdData and replace with what the customer needs to input.